bin/crew

```diff
@@ -1847,6 +1857,111 @@ def print_deps_tree(args)
+  abort "\nGITLAB_TOKEN environment variable not set.\n".lightred if ENV['GITLAB_TOKEN'].nil?
+
+  packages = ''
+  gitlab_token = ENV['GITLAB_TOKEN']
+
+  if ARGV.length > 0
+    ARGV.each do |package|
+      unless package == 'upload'
+        pkgfile = "#{CREW_LOCAL_REPO_ROOT}/packages/#{package}.rb"
+        if File.exist?(pkgfile)
+          packages += " #{package}"
+        else
+          puts "Package #{pkgfile} not found.".orange
+        end
+      end
+    end
+  else
+    %w[.tar.xz .tpxz .tar.zst].each do |ext|
+      packages += `find #{CREW_LOCAL_REPO_ROOT}/release/*/*#{ext} -exec basename -s #{ext} {} + | cut -d- -f1 | sort | uniq | xargs`
+      packages += ' '
+    end
+    abort 'No package binaries found.'.lightred if packages.empty?
+  end
+
+  base_url = 'https://gitlab.com/api/v4/projects/26210301/packages/generic'
+
+  packages.strip!
+
+  [packages].each do |package|
+    %w[x86_64 i686 armv7l].each do |arch|
+      puts "Package: #{package}, Arch: #{arch}".yellow
+      pkgfile = "#{CREW_LOCAL_REPO_ROOT}/packages/#{package}.rb"
+      new_tarfile = Dir["#{CREW_LOCAL_REPO_ROOT}/release/#{arch}/#{package}-*-chromeos-#{arch}.{tar.xz,tpxz,tar.zst}"].max_by { |f| File.mtime(f) }
+      if new_tarfile.nil?
+        puts "#{CREW_LOCAL_REPO_ROOT}/release/#{arch}/#{package}-#-chromeos-#{arch}.(tar.xz|tpxz|tar.zst) not found.\n".lightred
+        next
+      end
+      puts 'Generating sha256sum ...'
+      new_sha256 = `sha256sum #{new_tarfile} | cut -d' ' -f1`.chomp
+      puts "Uploading #{new_tarfile} ..."
+      noname = new_tarfile.split("#{package}-").last
+      new_version = noname.split('-chromeos').first
+      new_url = "#{base_url}/#{package}/#{new_version}_#{arch}/#{new_tarfile}".gsub("#{CREW_LOCAL_REPO_ROOT}/release/#{arch}/", '')
+      puts "curl -# --header \"DEPLOY-TOKEN: #{gitlab_token}\" --upload-file \"#{new_tarfile}\" \"#{new_url}\" | cat" if @opt_verbose
+      output = `curl -# --header "DEPLOY-TOKEN: #{gitlab_token}" --upload-file "#{new_tarfile}" "#{new_url}" | cat`.chomp
+      unless output.include?('201 Created')
+        puts output.lightred
+        puts "Unable to upload. Skipping binary_sha256 update in #{pkgfile}...".lightred
+        next
+      else
+        puts output.lightgreen
+      end
+      old_sha256 = `grep -m 1 #{arch}: #{pkgfile} 2> /dev/null`.chomp
+      unless old_sha256.empty?
+        old_sha256 = old_sha256.split("'")[1]
+        if old_sha256 != new_sha256
+          puts "Updating binary_sha256 in #{pkgfile}..."
+          puts "from: #{arch}: '#{old_sha256}'"
+          puts "  to: #{arch}: '#{new_sha256}'"
+          puts "sed -i 's/#{old_sha256}/#{new_sha256}/g' #{pkgfile}" if @opt_verbose
+          system "sed -i 's/#{old_sha256}/#{new_sha256}/g' #{pkgfile}"
+        else
+          puts "Skipping binary_sha256 update in #{pkgfile}..."
+        end
+      else
+        unless File.readlines("#{CREW_LOCAL_REPO_ROOT}/packages/#{package}.rb").grep(/binary_sha256/).any?
```

Comment:
Can't we use is_binary or something similar for this?

Reply:
No, this logic is here to check if the `binary_sha256` section exists and if not, inserts the lines in the package file.

Reply:
Ah, for packages which are lacking builds entirely.